hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/CacheWarmer.java

```diff
@@ -54,18 +55,17 @@ public class CacheWarmer {
     private final Executor executor;
 
     @Inject
-    public CacheWarmer(@NonNull final TransactionChecker checker, @NonNull final TransactionDispatcher dispatcher) {
-        this(checker, dispatcher, ForkJoinPool.commonPool());
-    }
-
-    @VisibleForTesting
-    CacheWarmer(
+    public CacheWarmer(
             @NonNull final TransactionChecker checker,
             @NonNull final TransactionDispatcher dispatcher,
-            @NonNull final Executor executor) {
+            @NonNull final ConfigProvider configProvider) {
         this.checker = checker;
         this.dispatcher = requireNonNull(dispatcher);
-        this.executor = requireNonNull(executor);
+        final int parallelism = configProvider
+                .getConfiguration()
+                .getConfigData(CacheConfig.class)
+                .cryptoTransferWarmThreads();
+        this.executor = new ForkJoinPool(parallelism);
     }
```

Comment:
I think it is better to pass the Executor in rather than the config. Using an executor makes testability better, because I can use a special testing executor that doesn't execute the task on the background thread until the test logic says it should, making it possible to test different concurrent scenarios directly. It seems like the Dependency Injection system could be the one that takes the config property and creates the FJP which will be injected into the CacheWarmer. Or if the CacheWarmer lifecycle is being managed outside of DI, then whoever manages its lifecycle could read the config value.